packages/eslint-plugin/docs/rules/README.md

```diff
@@ -6,18 +6,27 @@ pagination_prev: null
 slug: /
 ---
 
-`@typescript-eslint/eslint-plugin` includes over 100 rules that detect best practice violations, bugs, and/or stylistic issues specifically for TypeScript code.
-See [Configs](/linting/configs) for how to enable recommended rules using configs.
+## Introduction
 
-## Supported Rules
+`@typescript-eslint/eslint-plugin` includes over 100 rules that detect best practice violations, bugs, and/or stylistic issues specifically for TypeScript code. All of our rules are listed below.
 
-import RulesTable from "@site/src/components/RulesTable";
+Instead of enabling rules one by one, we recommend using one of [our pre-defined configs](/linting/configs) to easily enable a large set of
+recommended rules.
+
+## Rules
 
-<RulesTable ruleset="supported-rules" />
+The rules are listed in alphabetical order. You can optionally sort them based on these categories:
+
+import RulesTable from "@site/src/components/RulesTable";
 
-## Extension Rules
+<RulesTable />
 
-In some cases, ESLint provides a rule itself, but it doesn't support TypeScript syntax; either it crashes, or it ignores the syntax, or it falsely reports against it.
-In these cases, we create what we call an extension rule; a rule within our plugin that has the same functionality, but also supports TypeScript.
+## Sorting
 
-<RulesTable ruleset="extension-rules" />
+- "Config Group" refers to the configuration preset. We offer [three different config groups](/linting/configs) that allow users to enable a large set of recommended rules all at once.
+- `🔧 fixable` refers to whether the rule contains an ESLint auto-fixer. If the rule has an auto-fixer, then some rule violations can be fixed by running `eslint` with the `--fix` flag. This will automatically change the code, which can save a lot of time! (It is a common pattern for developers to configure their IDE to automatically run `eslint --fix` when saving a TypeScript file.)
+- `💡 has suggestions` refers to whether the rule will offer a suggestion of how to fix the code. Sometimes, it is not safe to automatically fix the code with an auto-fixer. But in these cases, we often have a good guess of what the correct fix should be, and we can provide it as a suggestion to the developer.
+- `💭 requires type information` refers to whether the rule needs to leverage the power of the TypeScript compiler in order to work properly. Rules that require type information are usually much slower than rules that don't. Thus, if linting performance becomes an issue in a gigantic codebase, a good first step might be make rules that require type information run only in CI.
+- `🧱 extension rule` means that the rule was originally a [core ESLint rule](https://eslint.org/docs/latest/rules/). Some core ESLint rules do not support TypeScript syntax: either they crash, ignore the syntax, or falsely report against it. In these cases, we create an "extension rule": a rule within our plugin that has the same functionality, but also supports TypeScript.
```

Comment:
Sometimes we do more than that: e.g. we use type information to refine the check. I think the only thing that's for sure is:

1. The rule either supports TypeScript syntax or leverages type informatoin
2. There exists a base rule that does similar stuff but in a less powerful way

Reply:
i didn't author this passage, it was taken from the existing page. if there are problems, do you want to suggest a commit?

Reply:
Ah, didn't realize that. Let's keep it as-is for now, then.

Reply:
Actually, where is it copied from? I don't find existing content that contains the same text.

Reply:
The main branch of this repo contains this text:

![image](https://github.com/typescript-eslint/typescript-eslint/assets/5511220/de958059-51fd-4ad4-9f42-0dc38ee4640f)